### PR TITLE
feat(helm): Adds a base helm chart for canary-checker

### PR DESCRIPTION
**Pull request by parth-gohil**

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    matchLabels:
+      {{- include "canary-checker.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "canary-checker.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "canary-checker.name" . }}-sa
+      volumes:
+        - name: dockersock
```
Comment (moshloop) [inline chart/templates/deployment.yaml:19]: Can we gate this behind a .Values.dockerSocket = true` ?

File: chart/templates/postgres.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+type: Opaque
+data:
+  {{- $secretObj := (lookup "v1" "Secret" .Release.Namespace "postgres-connection") | default dict }}
+  {{ $releaseNamespace := .Release.Namespace }}
+  {{- $secretData := (get $secretObj "data") | default dict }}
+  {{- $user := (get $secretData "POSTGRES_USER") | default ("postgres" | b64enc) }}
+  POSTGRES_USER: {{ $user | quote }}
+  {{- $password := (get $secretData "POSTGRES_PASSWORD") | default (randAlphaNum 32 | b64enc) }}
+  POSTGRES_PASSWORD: {{ $password | quote }}
+  {{- $canaryCheckerUrl := (get $secretData "DB_URL") | default ( ( print "postgresql://" $user ":" $password "@postgres." $releaseNamespace ".svc.cluster.local:5432/canary_checker" ) | b64enc ) }}
```
[the remark this answers is not in the document]
Comment (parth-gohil) [reply chart/templates/postgres.yaml:71]: Ya... I was thinking of the same thing... I started changing that but refrained due to some logic in my head...

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (parth-gohil) [reply chart/templates/deployment.yaml:20]: I've gated the volume creation and mounting of that volume.

File: Makefile [hunk 3]
```diff
@@ -241,3 +241,9 @@ bin: .bin .bin/wait4x .bin/yq .bin/karina .bin/go-junit-report .bin/restic .bin/
 # Generate all the resources and formats your code, i.e: CRDs, controller-gen, static
 .PHONY: resources
 resources: fmt static manifests
+
+.PHONY: chart
+chart:
```
[the remark this answers is not in the document]
Comment (parth-gohil) [reply Makefile:246]: In what way should the addition of crd be dependent on the manifest?

**Pull request merged**